docs/docs/docker/tagging-docker-images.mdx

```diff
@@ -131,6 +131,7 @@ docker_image(
 The default placeholders are:
 
 - `{directory}`: The directory the docker_image's BUILD file is in.
+- `{build_file_path}`: The full path to the BUILD file.
```

Comment:
iff we move forward with this patch, I'd prefer to call this `build_file_dir` to match the BUILD file function with the same name and purpose.

Reply:
:-) Apologies - I followed direction from slack - that it was build_file_path. 

![image](https://github.com/pantsbuild/pants/assets/1148383/c31593af-8c50-47c8-a7e1-68256fb8c5b6)

Reply:
also, I think we could re-order the items slightly, so we don't break up the `directory` and `parent_directory` pair, by moving this new one to either before or after those two.

Reply:
Actually on second thought, I'm not sure we should try to be close to `build_file_dir()` at all... as if this is used in a BUILD file it could look like:
```python
docker_image(
  repository="{build_file_dir}/some/example",
)
```
which looks like a typo for:
```python
docker_image(
  repository=f"{build_file_dir()}/some/example",
)
```

to avoid that, and align closer to the related `directory` and `parent_directory`, how do you guys feel about instead use `full_directory` or something along those lines?

... or maybe I'm overthinking this. 😬  Now I'm undecided what I think here. I'm fine with either `build_file_dir` or something completely different (that makes sense in context of the other available interpolation vars).

Reply:
@kaos you are not overthinking it at all.

That was close to what I originally had in mind.

full_path_directory <---
parent_directory
directory